Extensions/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/siteAliasSettings/siteAliases/index.jsx

```diff
@@ -141,7 +145,7 @@ class SiteAliasesPanel extends Component {
                         browser={item.BrowserType}
                         skin={item.Skin}
                         language={item.CultureCode}
-                        isPrimary={item.IsPrimary}
+                        isPrimary={!!item.IsPrimary}
```

Comment:
This does not seem correct.

Reply:
oh it's an old trick to convert to boolean

Reply:
Seems like a bit of a hack, but if that is considered best practice here, then fine.  Otherwise, I'd suggest tracing this down so that `item.IsPrimary` is never falsey.

Reply:
In minified javascript you'll notice many of these tricks. For instance, `true` is `!0`, and `false` is `!1`. But yeah, best to rely on type checking.

Reply:
> Approving so as not to block for the `!!` "trick" to coerce `item.IsPrimary` to a boolean if falsey. I do suggest this is eventually looked into though.

It also prevents changes from elsewhere to affect the code here ! inverterts the coalesced bool and ! reverts it back so we are sure that at least inside the component we get a real bool that way. 

The problem with languages that do not have strong typing is that even with propTypes and stuff, it just gives a warning, and sometimes only in production, so this protects a bit against bad usage in the api or the component that uses this one, etc.

Would typescript prevent passing a wrong type to a component or would it also just throw a warning that is easy to ignore ?